### ISSUE TITLE
Model breaks during presence of silence + speech

### ISSUE THREAD

**Issue by sahamrit** [guillaumekln/faster-whisper#322]
[Input Audio](https://drive.google.com/file/d/1oWchtlW9N8DaHV3VSKA6264-K4kcHpID/view?usp=drive_link)

**Config**
Large-v2, Beam=5, Language = en

**Desired Behaviour**

```
20:24.440 --> 20:31.880
easy, you have to... solve a problem.

20:31.880 --> 20:37.560
Solve problems.

20:37.560 --> 20:45.760
We have some difficult problems to solve.

```

**Actual Behaviour**

```
[20:19.840 --> 20:22.840]  but it's not easy, you have to...
[20:24.840 --> 20:26.840]  Solve a problem.
[20:26.840 --> 20:28.840]  Solve problems.
[20:28.840 --> 20:30.840]  Solve a problem.
[20:30.840 --> 20:32.840]  Solve problems.
[20:36.840 --> 20:39.840]  We have some difficult problems to solve.
```

**More Examples**

[Audio](https://www.youtube.com/watch?v=QkhPrdJEqgA)

First 3 minutes of this audio are completely broken. And again due to major silence regions.

**Desired Behaviour**

```
WEBVTT

00:30.000 --> 00:32.000
You

01:00.000 --> 01:02.000
you

01:02.000 --> 01:23.580
Hello, hello.

01:24.140 --> 01:26.140
Hello, hello

01:31.460 --> 01:33.460
Let's see

01:34.980 --> 01:39.300
This thing needs to go here

01:43.300 --> 01:45.300
It's a bit large

01:45.300 --> 01:51.460
Oh, that's better

01:51.460 --> 02:07.460
And this can probably be like this

02:22.460 --> 02:24.460
That's it

02:26.620 --> 02:32.020
Hello, will stability be involved? I don't I don't know not yet

02:33.100 --> 02:37.480
Which program is this for streaming? This is OBS. I think pretty much

02:38.260 --> 02:39.980
everyone

02:39.980 --> 02:41.700
uses OBS

02:41.700 --> 02:44.460
At least for YouTube. I think

02:44.980 --> 02:49.200
It's it's open the thing covered my message. Yes

02:49.200 --> 02:51.200
Sorry

02:53.520 --> 03:01.560
Talk about censoring and safety mechanism from what I've read. That's a major issue. I don't think it's a major issue

03:03.800 --> 03:07.640
So it's a major issue in in

03:08.760 --> 03:11.520
the open AI system, but
```

**Actual Behaviour**

```
[02:30.000 --> 02:33.900]  Silence.
[02:33.900 --> 02:37.100]  Silence.
[02:37.100 --> 02:40.300]  Silence.
[02:40.300 --> 02:43.480]  Silence.
[02:43.480 --> 02:46.500]  Silence.
[02:46.500 --> 02:49.460]  Silence.
[02:49.460 --> 02:52.440]  Silence.
[02:52.440 --> 02:55.420]  Silence.
[02:55.420 --> 02:58.440]  Silence.
[02:58.440 --> 03:11.560]  issue i don't think it's a a major issue um so it's a major issue in in the open ai system but

```

**Comment by sahamrit** [comment]
@guillaumekln  Analysed the output of encoders for this [audio](https://www.youtube.com/watch?v=QkhPrdJEqgA)

The output of encoder for the first segment that gets processed (0 to 3000 frames), is different from OpenAI implementation.

I have converted segment to fp16 explicitly 

`segment = segment.astype(np.float16)`

**Desired output - OpenAI codebase**

```
segment
tensor([[-0.5317, -0.5317, -0.5317,  ..., -0.5317, -0.5317, -0.5317],
        [-0.5317, -0.5317, -0.5317,  ..., -0.5317, -0.5317, -0.5317],
        [-0.5317, -0.5317, -0.5317,  ..., -0.5317, -0.5317, -0.5317],
        ...,
        [-0.5317, -0.5317, -0.5317,  ..., -0.5317, -0.5317, -0.5317],
        [-0.5317, -0.5317, -0.5317,  ..., -0.5317, -0.5317, -0.5317],
        [-0.5317, -0.5317, -0.5317,  ..., -0.5317, -0.5317, -0.5317]],
       device='cuda:0', dtype=torch.float16)
decode_output.audio_features
tensor([[-0.3940, -0.3433, -0.1383,  ..., -3.3770, -0.1664,  0.3040],
        [-0.0759,  0.0934,  0.1942,  ..., -3.8672, -0.6152,  0.0973],
        [ 0.5142,  0.2551,  1.1514,  ..., -3.8086, -0.5181,  0.3232],
        ...,
        [ 0.1667,  0.0173, -0.0627,  ..., -0.4285,  0.6802, -0.5718],
        [-0.2029,  0.6895, -0.6089,  ..., -0.5723, -0.2388, -0.5483],
        [ 0.2499,  0.3044, -0.6162,  ..., -0.0441,  1.0039, -0.3440]],
       device='cuda:0', dtype=torch.float16)
```

**Observed Output**

```
segment
array([[-0.5317, -0.5317, -0.5317, ..., -0.5317, -0.5317, -0.5317],
       [-0.5317, -0.5317, -0.5317, ..., -0.5317, -0.5317, -0.5317],
       [-0.5317, -0.5317, -0.5317, ..., -0.5317, -0.5317, -0.5317],
       ...,
       [-0.5317, -0.5317, -0.5317, ..., -0.5317, -0.5317, -0.5317],
       [-0.5317, -0.5317, -0.5317, ..., -0.5317, -0.5317, -0.5317],
       [-0.5317, -0.5317, -0.5317, ..., -0.5317, -0.5317, -0.5317]],
      dtype=float16)
output = torch.as_tensor(encoder_output,device="cuda")
output
tensor([[[-0.3953, -0.3440, -0.1372,  ..., -3.3770, -0.1687,  0.3008],
         [-0.0771,  0.0946,  0.1929,  ..., -3.8672, -0.6157,  0.1009],
         [ 0.5088,  0.2524,  1.1523,  ..., -3.8125, -0.5190,  0.3232],
         ...,
         [ 0.1707,  0.0165, -0.0626,  ..., -0.4326,  0.6792, -0.5698],
         [-0.2029,  0.6909, -0.6099,  ..., -0.5747, -0.2368, -0.5469],
         [ 0.2505,  0.3027, -0.6147,  ..., -0.0476,  1.0039, -0.3406]]],
       device='cuda:0', dtype=torch.float16)
```

**Comment by phineas-pta** [comment]
did u enable VAD ?

**Comment by sahamrit** [comment]
Thanks VAD needed to be enabled. I assumed it would have been default since OpenAI has it default too. My bad

**Comment by gordicaleksa** [comment]
Related to https://github.com/guillaumekln/faster-whisper/issues/183 opening this one until this is truly resolved.

The original implementation doesn't have a dedicated VAD filter, VAD filter slows down things and I've seen multiple issues people complaining about it.

**Comment by guillaumekln** [comment]
Thank you for bumping this issue @gordicaleksa. There was actually a bug in the application of `no_speech_threshold` which is fixed in #376. When the threshold was met for a segment, the next 30-second window reused the same encoder output and was also considered as non speech.

I just pushed version 0.7.1 with this fix.
